File: comdiv.py
from fractions import gcd
import sys
import math
input_data = sys.stdin.readlines()
del input_data[0]
for i in input_data:
	i = i.rstrip()
	a,b = i.split()
	g = gcd(int(a),int(b))
	sqt = math.sqrt(g)
	ans = 0
	i = 1
	while i <= sqt:
		if g % i == 0:
			ans += 2
			if i == g/i:
				ans = ans - 1
		i += 1
	print ans

File: comdiv.c
#include<stdio.h>
#include<math.h>

long int gcd(long int a,long int b){
	if (a%b == 0)
		return b;
	else
		return gcd(b,a%b);
}

int main(){
	long int i,t,a,b,j,g,ans;
	float sqt;
	scanf("%ld",&t);
	i = 1;
	while (i <= t){
		scanf("%ld %ld",&a,&b);
		if (a > b)
			g = gcd(a,b);
		else
			g = gcd(b,a);
		sqt = sqrt(g);
		ans = 0;
		j = 1;
		while (j <= sqt ){
			if (g % j == 0){
				ans = ans + 2;
				if (j == g/j)
					ans = ans - 1;
			}
			j += 1;
		}
		printf("%ld\n",ans);
		i += 1;

	}
	return 0;

}
